src/core/core.layouts.js

```diff
@@ -220,189 +381,85 @@ module.exports = {
 		// 9. Tell any axes that overlay the chart area the positions of the chart area
 
 		// Step 1
-		var chartWidth = width - leftPadding - rightPadding;
-		var chartHeight = height - topPadding - bottomPadding;
+		var chartWidth = width - padding.width;
+		var chartHeight = height - padding.height;
 		var chartAreaWidth = chartWidth / 2; // min 50%
 
-		// Step 2
-		var verticalBoxWidth = (width - chartAreaWidth) / verticalBoxes.length;
-
-		// Step 3
-		// TODO re-limit horizontal axis height (this limit has affected only padding calculation since PR 1837)
-		// var horizontalBoxHeight = (height - chartAreaHeight) / horizontalBoxes.length;
+		// Steps 2-3
+		var chartSize = {
+			w: width,
```

Comment:
It's not clear from the names `w` / `width` and `h` / `height` how the variables differ.

Maybe rename `w`  and `h` to `widthWithPadding` or `borderBoxWidth` or something?

Reply:
I realized that too. Should be clearer now.